readthedocs/rtd_tests/fixtures/spec/v2/schema.yml

```diff
@@ -0,0 +1,79 @@
+# Read the Docs configuration file
+
+# The version of the spec to be use
+version: enum('2')
+
+# Formats of the documentation to be built
+# Default: ['htmlzip', 'pdf', 'epub', 'singlehtmllocalmedia']
+formats: any(list(enum('htmlzip', 'pdf', 'epub', 'singlehtmllocalmedia')), enum('all'), required=False)
+
+# The path to the requirements file from the root of the project
+requirements_file: str(required=False)
+
+# Configuration for Conda support
+conda: include('conda', required=False)
+
+# Configuration for the documentation build process
+build: include('build', required=False)
+
+# Configuration of the Python executable to be used
+python: include('python', required=False)
+
+# Configuration for sphinx
+sphinx: include('sphinx', required=False)
+
+# Submodules configuration
+submodules: include('submodules', required=False)
+
+# Redirects for the current version to be built
+redirects: include('redirects', required=False)
+
+---
+
+conda:
+  # The path to the Conda environment file from the root of the project
+  file: str()
+
+build:
+  # The build docker image to be used
+  # Default: '2.0'
+  image: enum('1.0', '2.0', 'latest')
+
+python:
+  # The Python version
+  # Default: '2.7'
+  version: enum('2', '2.7', '3', '3.3', '3.4', '3.5', '3.6', required=False)
+
+  # Install the project using python setup.py install or pip
+  install: enum('pip', 'setup.py', required=False)
+
+  # Extra requirements sections to install in addition to the package dependencies
+  # Default: []
+  extra_requirements: list(str(), required=False)
+
+  # Give the virtual environment access to the global site-packages dir
+  # Default: false
+  system_packages: bool(required=False)
+
+sphinx:  
+  # The path to the conf.py file
+  # Default: rtd will try to find it
+  file: str(required=False)
+
+submodules:
+  # List of submodules to be included
+  # Default: []
+  include: any(list(str()), enum('all'), required=False)
+
+  # List of submodules to be ignored
+  # Default: []
+  exclude: any(list(str()), enum('all'), required=False)
```

Comment:
is it possible to mark these `include` and `exclude` options mutually exclusive?

Reply:
Nop, there isn't :/, I'll see if there is a way to implement a custom validator to do this.

Reply:
I imagine we'll probably need some kind double validation:

* First validation is whether it passes the schema
* Second will be whether the values are actually correct in RTD (eg. a redirect with a nonsense string, or mutually exclusive options (eg. `sphinx` & `mkdocs`) and similar. Something like field-level and model-level form validation.